server/src/validators/user.validators.ts

```diff
@@ -4,56 +4,47 @@ import { PrismaClient } from '@prisma/client';
 import { __PROD__ } from '../constants';
 import { FieldError } from '../models/error.model';
 
+// =============== Public Validators ===============
+
 /**
- * Checks to database to make sure that a username is not already taken,
- * this is typically run on a sign up request
- * @param username The username to search for
+ * Performs two checks on a given email address
+ * 1. Checks that the email matches a strong regex pattern for emails
+ * 2. Checks that the email is not already in use by another user
+ * @param email The email address to validate
  * @param prisma The prisma client
  * @param errors Any preexisting errors
- *
- * @returns Errors after the validation
  */
-export const validateUserDoesNotExist = async (
-	username: string,
+export const validateNewEmail = async (
+	email: string,
 	prisma: PrismaClient,
 	errors: FieldError[] = [],
 ): Promise<FieldError[]> => {
-	// Ensures that the username isn't taken
-	const usernameIsTaken =
-		(await prisma.user.findUnique({
-			where: { username },
-		})) !== null;
+	// Gets the errors array after running the pattern match
+	const patternErrors = validateEmailPattern(email);
 
-	if (usernameIsTaken) {
-		errors.push({
-			field: 'username',
-			message: `The username '${username}' is taken`,
-		});
+	if (patternErrors.length > 0) {
+		// Returns the error array with the additional pattern error
+		return [...errors, ...patternErrors];
```

Comment:
This is primarily done for code flexibility if I decide that pattern errors can have more than 1 error. Otherwise push would work